### ISSUE TITLE
Shadow DOM: Field fails to detect presence of error text

### ISSUE THREAD

**Issue by mbeckem** [chakra-ui/ark#3485]
### Description

Hi,

I am using Ark via Chakra v3 in a Shadow Root. I noticed that input elements in `<Field.Root />` will not link to their `<Field.ErrorText />` description via the `aria-describedby` attribute (hurting usability when using a screen reader). This works fine without a Shadow DOM, as can be seen in [this example](https://www.chakra-ui.com/docs/components/field#error-text).

### Link to Reproduction (or Detailed Explanation)

See below

### Steps to Reproduce

1. Use the simple example from the documentation within a shadow DOM:

    ```tsx
    import { Field, Input } from "@chakra-ui/react";

    <Field.Root invalid>
        <Field.Label>Email</Field.Label>
        <Input placeholder="me@example.com" />
        <Field.ErrorText>This is an error text</Field.ErrorText>
    </Field.Root>
    ```
2. Inspect the `input` element in the browser, notice that the `aria-describedby` attribute is missing. 
   Compare this with the [working example](https://www.chakra-ui.com/docs/components/field#error-text) in Chakra's documentation .

   The `Field.Root` component fails to detect the presence of the `Field.ErrorText` child.

### Ark UI Version

5.9.1

### Framework

- [x] React
- [x] Solid
- [x] Vue

### Additional Information

I've tracked this down to the following dom query in `use-field.ts`: https://github.com/chakra-ui/ark/blob/596231c4d8b95e223374c54a8de96313ca6493c7/packages/react/src/components/field/use-field.ts#L67C1-L70C6

The items are queried using `document.getElementById()`, which will not find the children in the shadow DOM. Instead, the lookup should take into account the root node returned by the environment context:

```ts
const { getRootNode } = useEnvironmentContext();
// basically getRootNode().getElementById(errorTextId), but the reality is a bit more complicated
// since getRootNode()'s return value is very generic.
```

Looks like this problem also occurs in [useFieldset()](https://github.com/chakra-ui/ark/blob/596231c4d8b95e223374c54a8de96313ca6493c7/packages/react/src/components/fieldset/use-fieldset.ts#L42C1-L45C6).

**Comment by pgonzalez-santiago** [comment]
Hi @segunadebayo, I’m running into an issue related to this. In my case, `getElementById` doesn’t exist on `docOrShadowRoot`, and it’s causing an error. I noticed that you added [this change](https://github.com/chakra-ui/ark/commit/26ef3888c2d2606c1b345d775eff133d68b7f933#diff-05927c7601a2eb9de5f57d36df417269cc7d36ebf9f36f57d8a1fc4b28ec0698R68), and with the type assertion, the third possibility for a RootNode = Node (`type RootNode = ShadowRoot | Document | Node;`) was omitted. By leaving it out, this potential issue has been hidden. Was there a particular reason for this?

Edit: The problem was finally in my code, since I had initialized the [EnvironmentProvider](https://ark-ui.com/docs/utilities/environment) incorrectly: `<EnvironmentProvider value={() => element.current?.shadowRoot ?? document.body>` instead of `<EnvironmentProvider value={() => element.current?.shadowRoot ?? document}>`
